### PR TITLE
chore(samples): update Autoclass Sample for v2.1 features

### PR DESCRIPTION
**Pull request by BenWhitehead**
Depends on https://github.com/googleapis/java-storage/pull/2258

### CODE REVIEW

**Review by lesv — APPROVED** [review]
No test?  Otherwise LGTM

File: samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java [hunk 1]
```diff
@@ -21,32 +21,54 @@
...
-    // To enable autoclass, you must enable it at bucket creation time.
+    // The storage class that objects in an Autoclass bucket eventually transition to if not read
+    // for a certain length of time
+    // StorageClass storageClass = StorageClass.NEARLINE;
+
+    // Configure the Autoclass setting for a bucket.
+
+    // Note:  terminal_storage_class field is optional and defaults to NEARLINE if not otherwise
+    // specified. Valid terminal_storage_class values are NEARLINE and ARCHIVE.
     boolean enabled = false;
```
Comment (cojenco) [inline samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java:45]:
```suggestion
    boolean enabled = true;
```

File: samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java [hunk 2]
```diff
@@ -21,32 +21,54 @@
...
 
     // The ID of your GCS bucket
     // String bucketName = "your-unique-bucket-name";
 
-    // Whether to set Autoclass to on or off.
-    // Note: Only update requests that disable qutoclass are currently supported.
-    // To enable autoclass, you must enable it at bucket creation time.
+    // The storage class that objects in an Autoclass bucket eventually transition to if not read
+    // for a certain length of time
+    // StorageClass storageClass = StorageClass.NEARLINE;
```
Comment (cojenco) [inline samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java:39]:
```suggestion
    // StorageClass storageClass = StorageClass.ARCHIVE;
```

Comment (cojenco) [reply samples/snippets/src/main/java/com/example/storage/bucket/SetBucketAutoclass.java:39]: Apologies for the canonical sample changes, let's update this sample to set `enabled = true`  with `storageClass = StorageClass.ARCHIVE`. This is more aligned with the CUJ for setting terminal storage class. Thank you!

**Review by cojenco — COMMENTED** [review]
Thanks for working on this! Could we also update the storage_get_autoclass sample to demonstrate the newly added fields?

https://github.com/googleapis/java-storage/blob/main/samples/snippets/src/main/java/com/example/storage/bucket/GetBucketAutoclass.java

**Comment by BenWhitehead** [comment]
Now that libraries-bom has been released with the feature I've rebased and updated the sample and the samples test.

@cojenco I applied the value changes in the sample you recommended.

**Comment by cojenco** [comment]
> Thanks for working on this! Could we also update the storage_get_autoclass sample to demonstrate the newly added fields?
> 
> https://github.com/googleapis/java-storage/blob/main/samples/snippets/src/main/java/com/example/storage/bucket/GetBucketAutoclass.java

LGTM, could we showcase the new fields in the get_autoclass sample too? Thank you @BenWhitehead

**Review by cojenco — APPROVED** [review]
lgtm, thanks Ben 🙂

**Pull request merged**